### ISSUE TITLE
Dart layer - Format on save does not always work (indentation)

### ISSUE THREAD

**Issue by nicobao** [syl20bnr/spacemacs#13566]
#### Description :octocat:
Layer: dart using lsp (see config below - I also add manually lsp-dart to additional-packages or else I get an error).
When editing .dart file, format-on-save does not systematically work. Sometimes it does, sometimes it doesn't.

![format-on-save-dart-fails](https://user-images.githubusercontent.com/46368291/81349859-23af4b80-90c1-11ea-94d1-436eca942358.gif)

#### Reproduction guide :beetle:
- Start Emacs
- Open a well-formatted dart file (dartfmt has already been run on the file)
- Arbitrarily indent the code with `CTRL->` somewhere (evil mode)
- Save with `:w`  (evil mode)

*Observed behaviour:* :eyes: :broken_heart:
The file is not automatically formatted back using dartfmt.
The new ugly indent still remains.

*Expected behaviour:* :heart: :smile:
The file should be automatically formatted using dartfmt.
The new ugly indent should be removed.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.91
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 1f6e39ea8)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(yaml sql
      (html :variables web-fmt-tool 'web-beautify)
      react
      (auto-completion :variables auto-completion-return-key-behavior nil auto-completion-tab-key-behavior 'complete)
      prettier latex
      (dart :variables dart-backend 'lsp lsp-dart-sdk-dir "/opt/flutter/bin/cache/dart-sdk/" lsp-enable-on-type-formatting t)
      (javascript :variables javascript-backend 'lsp javascript-lsp-linter nil javascript-import-tool 'import-js)
      emacs-lisp git ivy erlang lsp
      (elixir :variables elixir-backend 'lsp)
      markdown multiple-cursors org python
      (shell :variables shell-default-height 30 shell-default-position 'bottom)
      (spell-checking :variables spell-checking-enable-by-default nil)
      sphinx restructuredtext syntax-checking version-control treemacs)
```
- System configuration features: XPM JPEG TIFF GIF PNG SOUND DBUS GSETTINGS GLIB NOTIFY INOTIFY LIBSELINUX GNUTLS FREETYPE HARFBUZZ XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS PDUMPER GMP

```bash
[nicolas@linux-1 example]$ /opt/flutter/bin/cache/dart-sdk/bin/dartfmt --version
1.3.2
[nicolas@linux-1 example]$ /opt/flutter/bin/cache/dart-sdk/bin/dart --version
Dart VM version: 2.7.0 (Fri Dec 6 16:26:51 2019 +0100) on "linux_x64"
```

**Comment by lebensterben** [comment]
@nicobao 
Is it fixed after `lsp-dart` 1.15?

ref: https://github.com/emacs-lsp/lsp-dart/blob/49bb4d597e30c95d66a96739ea10c1c4f74ac649/CHANGELOG.md#1150

**Comment by nicobao** [comment]
Hi,
Thanks for your reply.
I don't know because I haven't used spacemacs nor dart for quite a long time now.
